docs/pages/inboxes/disappearing-messages.md

```diff
@@ -0,0 +1,97 @@
+# Support disappearing messages with XMTP
+
+Disappearing messages are messages that are intended to be visible to users for only a short period of time. After the message expiration time passes, the messages are removed from the UI and deleted from local storage so the messages are no longer accessible to conversation participants.
+
+Disappearing message behavior is enforced by apps, meaning that apps are responsible for removing messages from their UIs and local storage based on conditions set at the conversation level. Conversation participants using apps that support disappearing messages will have a UX that honors the message expiration conditions. 
+
+However, it's important to understand that:
+
+- A conversation participant using an app that doesn't support disappearing messages will not experience disappearing message behavior.
+- Messages aren't deleted from the XMTP network.
+
+Therefore, disappearing messages should be understood as best-effort, app-level privacy that helps avoid leaving an easily accessible record in a messaging UI. However, it's not a guaranteed, system-wide “message self-destruct.”
+
+## Implement disappearing messages
+
+Here is a high-level view of the disappearing messages flow:
+
+1. When a 1:1 or group chat conversation is created or updated, a user with appropriate permissions can set disappearing message conditions for messages sent in the conversation.
+2. In a conversation with disappearing message conditions, a sender creates a message. The message abides by disappearing message metadata set for the conversation.
+3. The message is sent over the XMTP network.
+4. The recipients’ apps with disappearing message support receive the message and check the expiration conditions set for the conversation.
+5. If the message hasn't expired, the app displays the message in the messaging UI.
+6. If the message has expired, the message is removed from (or not displayed in the first place in) the conversation participants’ app UIs and is deleted from the apps’ local storage.
+
+### Enable disappearing messages for a conversation
+
+When creating or updating a conversation, enable users with appropriate permissions to set disappearing message expiration conditions.
+
+This includes setting the following conditions expressed in nanoseconds (ns):
+
+- `from_ns`: Starting timestamp from which the message lifespan is calculated
+- `in_ns`: Duration of time during which the message should remain visible to conversation participants
+
+For example:
```

Comment:
Did I get this correct, @mchenani? 🫣 😆

Reply:
I'd say lets wait for the specific code snippets till it's out in the sdks since things will be differently exposed.